docs/rules/no-potentially-empty-backreference.md

```diff
@@ -0,0 +1,61 @@
+---
+pageClass: "rule-details"
+sidebarDepth: 0
+title: "regexp/no-potentially-empty-backreference"
+description: "disallow backreferences that reference a group that might not be matched"
+---
+# regexp/no-potentially-empty-backreference
+
+> disallow backreferences that reference a group that might not be matched
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+
+## :book: Rule Details
+
+If the referenced group of a backreference is not matched because some other
+path leads to the backreference, the backreference will be replaced with the
+empty string. The same will happen if the captured text of the referenced group
+was reset before reaching the backreference.
+
+This will handle backreferences which will always be replaced with the empty
+string for the above reason. Use [regexp/no-useless-backreference] for that.
```

Comment:
```suggestion
If the referenced group of a backreference is not matched because some other path leads to the backreference, the backreference will trivially accept (e.g. `/(?:(a)|b)\1/`). The same will happen if the captured text of the referenced group was reset before reaching the backreference.

This will not handle backreferences that always trivially accept. Use [regexp/no-useless-backreference] for that.
```